presto-hive-hadoop2/src/test/java/io/prestosql/plugin/hive/TestHiveHadoop2Plugin.java

```diff
@@ -99,6 +103,69 @@ public void testGcsAccessTokenAndHiveCachingMutuallyExclusive()
                 .hasMessageContaining("Use of GCS access token is not compatible with Hive caching");
     }
 
+    @Test
+    public void testImmutablePartitionsAndInsertOverwriteMutuallyExclusive()
+    {
+        Plugin plugin = new HiveHadoop2Plugin();
+        ConnectorFactory connectorFactory = Iterables.getOnlyElement(plugin.getConnectorFactories());
+
+        assertThatThrownBy(() -> connectorFactory.create(
+                "test",
+                ImmutableMap.<String, String>builder()
+                        .put("hive.insert-existing-partitions-behavior", "APPEND")
+                        .put("hive.immutable-partitions", "true")
+                        .put("hive.metastore.uri", "thrift://foo:1234")
+                        .build(),
+                new TestingConnectorContext()))
+                .hasMessageContaining("Presto is configured to treat Hive partitions as immutable. insert_existing_partitions_behavior is not allowed to be set to APPEND");
+    }
+
+    @Test
+    public void testInsertOverwriteIsSetToErrorWhenImmutablePartitionsIsTrue()
+    {
+        Plugin plugin = new HiveHadoop2Plugin();
+        ConnectorFactory connectorFactory = Iterables.getOnlyElement(plugin.getConnectorFactories());
+
+        Connector connector = connectorFactory.create(
+                "test",
+                ImmutableMap.<String, String>builder()
+                        .put("hive.immutable-partitions", "true")
+                        .put("hive.metastore.uri", "thrift://foo:1234")
+                        .build(),
+                new TestingConnectorContext());
+
+        Object actual = getDefaultValueInsertExistingPartitionsBehavior(connector);
+
+        assertEquals(ERROR, actual);
+    }
+
+    @Test
+    public void testInsertOverwriteIsSetToAppendWhenImmutablePartitionsIsFalseByDefault()
+    {
+        Plugin plugin = new HiveHadoop2Plugin();
+        ConnectorFactory connectorFactory = Iterables.getOnlyElement(plugin.getConnectorFactories());
+
+        Connector connector = connectorFactory.create(
+                "test",
+                ImmutableMap.<String, String>builder()
+                        .put("hive.metastore.uri", "thrift://foo:1234")
+                        .build(),
+                new TestingConnectorContext());
+
+        Object actual = getDefaultValueInsertExistingPartitionsBehavior(connector);
+
+        assertEquals(APPEND, actual);
+    }
+
+    private Object getDefaultValueInsertExistingPartitionsBehavior(Connector connector)
```

Comment:
nit: you can use concrete type here

Reply:
It requires unnecessary explicit casting. I think it does not worth adding additional characters in this case. 
However I would be glad to substitute method's body with shorter and more understandable implementation, but I can't find one.